_pytest/python.py

```diff
@@ -32,6 +32,26 @@
 # The type of re.compile objects is not exposed in Python.
 REGEX_TYPE = type(re.compile(''))
 
+
+if hasattr(inspect, 'signature'):
+    def _has_positional_arg(func):
+        sig = inspect.signature(func)
+        params = list(sig.parameters.values())
+        if len(params):
+            return params[0].kind in (sig.POSITIONAL_ONLY, sig.POSITIONAL_OR_KEYWORD)
+        else:
+            return False
+
+    def _format_args(func):
+        return str(inspect.signature(func))
+else:
+    def _has_positional_arg(func):
+        return inspect.getargspec(func)[0] is not None
```

Comment:
I think `.getargspec(...)[0]` will never be `None`, it seems to be the empty list instead:

```python
>>> def foo(): pass
... 
>>> inspect.getargspec(foo)
ArgSpec(args=[], varargs=None, keywords=None, defaults=None)
```

I think this should be `bool(inspect.getargspec(func).args)` instead. It's a named tuple since 2.6, so why not use that too. :wink:f